iothub/device/src/Authentication/AuthenticationMethodFactory.cs

```diff
@@ -10,7 +10,7 @@ namespace Microsoft.Azure.Devices.Client
     /// </summary>
     public sealed class AuthenticationMethodFactory
     {
-        internal static IAuthenticationMethod GetAuthenticationMethod(IotHubConnectionStringBuilder csBuilder)
```

Comment:
This method gets called only from the connection string flow, so renaming it to make it more specific